File: frontend/src/types/pubs.ts
export interface PubInfo {
  id: number;
  merchant_user_id: number;
  pub_name: string;
  cover_url: string | null;
  address: string | null;
  longitude: number | null;
  latitude: number | null;
  contact_phone: string | null;
  business_hours: string | null;
  status: number;
}

export interface BeerPubItem {
  pub: PubInfo;
  beer_name: string;
  brewery_name: string | null;
  style?: string | null;
  price: number | null;
  status: number;
  distance_km?: number;
}

export interface BeerPubSearchResponse {
  items: BeerPubItem[];
  offset: number;
  limit: number;
}

export interface BeerNearbyQuery {
  beer_name: string;
  lat: number;
  lng: number;
  radius_km: number;
  in_stock_only?: boolean;
  offset?: number;
  limit?: number;
}

export interface PubRouteResponse {
  mode: "walking" | "driving";
  distance_m: number;
  duration_min: number;
  origin: { lng: number; lat: number };
  destination: { lng: number; lat: number };
}

export interface PubNearbyItem {
  pub: PubInfo;
  distance_km: number;
}

export interface PubNearbyResponse {
  items: PubNearbyItem[];
  total: number;
}

export interface PubListResponse {
  items: PubInfo[];
  offset: number;
  limit: number;
}

export interface PubNearbyQuery {
  lat: number;
  lng: number;
  radius_km?: number;
  offset?: number;
  limit?: number;
}

export interface PubUpdatePayload {
  pub_name?: string;
  cover_url?: string;
  address?: string;
  city?: string;
  longitude?: number;
  latitude?: number;
  contact_phone?: string;
  business_hours?: string;
  status?: number;
}

File: src/schema/pubs.py
from typing import Optional, List

from pydantic import BaseModel


class PubCreateRequest(BaseModel):
    pub_name: str
    cover_url: Optional[str] = None
    address: str
    city: Optional[str] = None
    longitude: Optional[float] = None
    latitude: Optional[float] = None
    contact_phone: Optional[str] = None
    business_hours: Optional[str] = None
    status: int = 1


class PubUpdateRequest(BaseModel):
    pub_name: Optional[str] = None
    cover_url: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    longitude: Optional[float] = None
    latitude: Optional[float] = None
    contact_phone: Optional[str] = None
    business_hours: Optional[str] = None
    status: Optional[int] = None


class PubResponse(BaseModel):
    id: int
    merchant_user_id: int
    pub_name: str
    cover_url: Optional[str] = None
    address: Optional[str] = None
    longitude: Optional[float] = None
    latitude: Optional[float] = None
    contact_phone: Optional[str] = None
    business_hours: Optional[str] = None
    status: int


class PubListResponse(BaseModel):
    items: List[PubResponse]
    offset: int
    limit: int


class BeerPubItem(BaseModel):
    pub: PubResponse
    beer_name: str
    brewery_name: str
    style: Optional[str] = None
    price: Optional[float] = None
    status: int
    distance_km: Optional[float] = None


class BeerPubSearchResponse(BaseModel):
    items: List[BeerPubItem]
    offset: int
    limit: int


class PubNearbyItem(BaseModel):
    pub: PubResponse
    distance_km: float


class PubNearbyResponse(BaseModel):
    items: List[PubNearbyItem]
    total: int
